python_modules/libraries/dagster-looker/dagster_looker/asset_utils.py

```diff
@@ -24,12 +24,17 @@ def build_looker_dashboard_specs(
     # https://cloud.google.com/looker/docs/reference/param-lookml-dashboard
     for lookml_dashboard_path in project_dir.rglob("*.dashboard.lookml"):
         for lookml_dashboard_props in yaml.safe_load(lookml_dashboard_path.read_bytes()):
+            lookml_dashboard = (lookml_dashboard_path, lookml_dashboard_props)
+
             looker_dashboard_specs.extend(
                 AssetSpec(
-                    key=dagster_looker_translator.get_asset_key(
-                        lookml_element=(lookml_dashboard_path, lookml_dashboard_props)
-                    ),
+                    key=dagster_looker_translator.get_asset_key(lookml_dashboard),
                     deps={AssetKey(["explore", lookml_dashboard_element_props["explore"]])},
+                    description=dagster_looker_translator.get_description(lookml_dashboard),
```

Comment:
Having to pass each of these in every AssetSpec construction callsite is a little rough. It also means we need to define a new method in `DagsterLookerTranslator` for each property type. This feels like something that `AssetSpec` is built to handle

One alternative path that comes to mind is something like

```python
customize_asset_spec(
  self, 
  asset_spec: AssetSpec,
  lookml_element: Tuple[Path, Mapping[str, Any]]
) -> AssetSpec:
  pass
```

For example

```python
class CustomDagsterLookerTranslator(DagsterLookerTranslator):
  
  customize_asset_spec(
    self, 
    asset_spec: AssetSpec,
    lookml_element: Tuple[Path, Mapping[str, Any]]
  ) -> AssetSpec:
    return asset_spec._replace(description=expected_description)
```

Or, if we added first-class "replacers" to `AssetSpec`:

```python
class CustomDagsterLookerTranslator(DagsterLookerTranslator):
  
  customize_asset_spec(
    self, 
    asset_spec: AssetSpec,
    lookml_element: Tuple[Path, Mapping[str, Any]]
  ) -> AssetSpec:
    return asset_spec.with_description(expected_description)
```

This feels a little better for me in a few ways:

1) If we add a new property to `AssetSpec`, we only need to add a new "replacer" to `AssetSpec` - no change to integration needed. With the current setup, we need to update all the `AssetSpec` construction sites here and also add a new method to `DagsterLookerTranslator`.

2) Users can consolidate any shared logic (calculating interim values or extracting parts of the lookml element) between their `DagsterLookerTranslator` getters, since they only have to implement a single method.

I'm curious if we've had this conversation w/r/t the dbt translator

Reply:
Also fine with keeping with this pattern for now, but curious for your thoughts

Reply:
I think we can keep the existing pattern, since it follows what we have in dbt and dlt. 

I agree that we can now consolidate this pattern to a function that returns a `AssetSpec` object. We could probably do that for all of our integrations in one go, once we're ready.